EarablesKIT/EarablesKIT/EarablesKIT/ViewModels/MusicModeViewModel.cs

```diff
@@ -1,31 +1,132 @@
-﻿using EarablesKIT.Models.Extentionmodel.Activities;
+﻿using EarablesKIT.Models;
+using EarablesKIT.Models.Extentionmodel;
+using EarablesKIT.Models.Extentionmodel.Activities;
+using EarablesKIT.Models.Extentionmodel.Activities.RunningActivity;
+using EarablesKIT.Models.Extentionmodel.Activities.StepActivity;
+using EarablesKIT.Models.Library;
+using MediaManager;
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
 using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Forms;
 
 namespace EarablesKIT.ViewModels
 {
-    class MusicModeViewModel : BaseModeViewModel
+    class MusicModeViewModel : BaseModeViewModel, INotifyPropertyChanged
     {
+        private bool _running = false;
+        private bool _musicModeActive = false;
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+
+        public bool IsRunning
+        {
+            set
+            {
+                _running = value;
+                if (_running)
+                {
+                    CrossMediaManager.Current.Play();
+                } else
+                {
+                    CrossMediaManager.Current.Pause();
+                }
+                OnPropertyChanged("StartStopLabel");
+            }
+            get => _running;
+        }
+
+        public Command ToggleMusicMode
+        {
+            get => new Command(() =>
+            {
+                _musicModeActive = !_musicModeActive;
+                if (_musicModeActive)
+                {
+                    StartActivity();
+                } else
+                {
+                    StopActivity();
+                }
+                //CrossMediaManager.Current.PlayPause();
+                OnPropertyChanged(nameof(StartStopLabel));
+            });
+        }
+
+        
+        public string StartStopLabel
+        {
+            get => _musicModeActive ? "Stop" : "Start";
+        }
+
+        private async void InitMusic(string urlPath)
+        {
+            await CrossMediaManager.Current.Play(urlPath);
+            await CrossMediaManager.Current.Pause();
+        }
+
+        private IActivityManager _activityManager;
+        private AbstractRunningActivity _runningActivity { get; set; }
 
         public MusicModeViewModel()
         {
-            throw new NotImplementedException();
+            InitMusic("https://sampleswap.org/samples-ghost/PUBLIC%20DOMAIN%20MUSIC/3337[kb]Extracts-from-the-Ballet-Suite-Scherazada.mp3.mp3");
```

Comment:
so wird die musik jedes mal neu runtergeladen. d.h. man brauch internet. Es sollte eine mp3 o.ä. in das projekt eingebunden sein, sodass das feature auch offline funktioniert.